web/app/themes/justicejobs/src/js/modal.js

```diff
@@ -0,0 +1,180 @@
+    console.log(firstPopupButton);
+    firstPopupButton.focus();
+    var iframe = document.querySelector('iframe');
+    if (iframe) {
+      var iframeSrc = iframe.src;
+      iframe.src = iframeSrc;
+    }
+  });
+
+  $(document).on('keydown', function (event) {
+    if($('.popup.is-opened')) {
+      if (event.key == "Escape" || event.keyCode === 27) {
+        $('.popup .btn-close').click();
+        var firstPopupButton = document.getElementsByClassName("carousel-popup-open")[0];
+        firstPopupButton.focus();
+      }
+    }
+  });
+
+  // Closes popup if background is clicked
+  $(".popup").click(function (event) {
+    if ($(event.target).hasClass('is-opened')) {
+      $('.popup').removeClass('is-opened');
+      enableBodyScrolling();
+      var firstPopupButton = document.getElementsByClassName("carousel-popup-open")[0];
+      firstPopupButton.focus();
+    }
+  });
+});
+
+
+function maybe_show_close_btn(ele) {
+  if (ele instanceof jQuery && ele.length > 0) {
+    var closeBtn = ele.find('.btn-close'),
+      position = closeBtn.position(),
+      btnHeight = closeBtn.outerHeight(),
+      ii = 100,
+      cnt = 0;
+
+    if (!position.top) {
+      return false;
+    }
+
+    if (closeBtn.is(':offscreen') || (position.top - 5) < btnHeight) {
+      var blockToMove = ele.find('.popup__block'),
+        distance = 10;
+
+      for (ii; cnt < ii; ii++) {
+        distance = distance + 20;
+        blockToMove.css({
+          top: distance
+        });
+
+        if (!closeBtn.is(':offscreen')) {
+          blockToMove.css({
+            top: distance + btnHeight
+          });
+          break;
+        }
+        cnt++;
+      }
+    }
+  }
+}
+
+/**
+ * Create a new jquery selector - :offscreen
+ * obj.is(':offscreen')
+ * @param el
+ * @returns {boolean}
+ */
+
+jQuery.expr.filters.offscreen = function (el) {
+  var rect = el.getBoundingClientRect();
+  return (
+    (rect.x + rect.width) < 0
+    || (rect.y + rect.height) < 0
+    || (rect.x > window.innerWidth || rect.y > window.innerHeight)
+  );
+};
+
+function disableBodyScrolling() {
+  document.body.style.position = 'fixed';
+  document.body.style.top = `-${window.scrollY}px`;
+}
+
+function enableBodyScrolling() {
+  const scrollY = document.body.style.top;
+  document.body.style.position = '';
+  document.body.style.top = '';
+  window.scrollTo(0, parseInt(scrollY || '0') * -1);
+}
+
+
+// Based on example by Hidde de Vries
+// https://hiddedevries.nl/en/blog/2017-01-29-using-javascript-to-trap-focus-in-an-element
+
+function trapFocus() {
+  var close = $('.popup.is-opened .btn-close');
+  close.focus();
+  var element = document.querySelector('.popup.is-opened');
+  var focusableEls = element.querySelectorAll('.popup.is-opened a[href]:not([disabled]), button:not([disabled]');
+  var firstFocusableEl = focusableEls[0];
+  lastFocusableEl = focusableEls[focusableEls.length - 1];
+  KEYCODE_TAB = 9;
```

Comment:
`lastFocusableEl` and `KEYCODE_TAB` have both been declared as global variables. Unless they are used outside the function it would be best to keep them organised in `document` as opposed to directly accessing off of `window`

Reply:
Good spot - have made them local to that function :)